### PR TITLE
Move Ersed banner logic to backend with ERSED<CRD<MTD

### PR DESCRIPTION
**Pull request by FaizJD030**
This is a bug fix for https://dsdmoj.atlassian.net/browse/CRS-1967
Move the ERSED N/A banner logic to the backend.

The FE will show the banner based on the ersedNotApplicableDueToDtoLaterThanCrd flag.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt [hunk 1]
```diff
@@ -388,7 +389,7 @@ class BookingExtractionService(
     dates: MutableMap<ReleaseDateType, LocalDate>,
     latestReleaseDate: LocalDate,
     sentenceGroups: List<List<CalculableSentence>>,
-  ) {
+  ): Boolean {
```
Comment (james-baker-digital-justice) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt:392]: Maybe update this name as it's not clear what the boolean is returning without looking at the call site.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt [hunk 2]
```diff
@@ -423,20 +425,22 @@ class BookingExtractionService(
...
       }
     }
+    return false
   }
 
   private fun isTusedableDtos(booking: Booking, effectiveSentenceLength: Period): Boolean {
     return booking.sentences.all { it.isDto() } && effectiveSentenceLength.toTotalMonths() < 24 && !booking.underEighteenAtEndOfCustodialPeriod()
   }
 
-  private fun calculateErsedWhereDtoIsPresent(
```
Comment (james-baker-digital-justice) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt:435]: why is this no longer private?

**Review by james-baker-digital-justice — CHANGES_REQUESTED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt (hunk 2 above)
Comment (FaizJD030) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt]: It was a typo, reverted.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt (hunk 1 above)
Comment (FaizJD030) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt:392]: The name is updated to reflect what the function does - extractErsedAndNotApplicableDueToDtoLaterThanCrdFlag

**Review by james-baker-digital-justice — APPROVED** [review]
LGTM

**Pull request merged**